stable_baselines3/common/distributions.py

```diff
@@ -292,6 +292,112 @@ def log_prob(self, actions: th.Tensor) -> th.Tensor:
         return self.distribution.log_prob(actions)
 
 
+class MultiCategoricalDistribution(Distribution):
+    """
+    MultiCategorical distribution for multi discrete actions.
+
+    :param action_dims: ([int]) List of sizes of discrete action spaces.
+    """
+    def __init__(self, action_dims: List[int]):
+        super(MultiCategoricalDistribution, self).__init__()
+        self.action_dims = action_dims
+        self.distributions = None
+
+    def proba_distribution_net(self, latent_dim: int) -> nn.Module:
+        """
+        Create the layer that represents the distribution:
+        it will be the logits of the MultiCategorical distribution.
+        You can then get probabilities using a softmax.
+
+        :param latent_dim: (int) Dimension of the last layer
+            of the policy network (before the action layer)
+        :return: (nn.Linear)
+        """
+        action_logits = nn.Linear(latent_dim, sum(self.action_dims))
+        return action_logits 
+
+    def proba_distribution(self, action_logits: th.Tensor) -> 'MultiCategoricalDistribution':
+        reshaped_logits = th.split(action_logits, self.action_dims, dim=1)
+        self.distributions = [Categorical(logits=l) for l in reshaped_logits]
+        return self
+
+    def mode(self) -> th.Tensor:
+        return th.stack([th.argmax(d.probs, dim=1) for d in self.distributions])
+
+    def sample(self) -> th.Tensor:
+        return th.stack([d.sample() for d in self.distributions])
+
+    def entropy(self) -> th.Tensor:
+        return sum([d.entropy() for d in self.distributions])
```

Comment:
this need to be double checked... as it will always return a scalar (not sure if this is the expected behavior, please check with `Categorical` what shape is returned).

Reply:
<class 'torch.Tensor'>
torch.Size([5000000])

I'm seeing the expected returns, no? Same returns as Categorical. I was having problems with th.sum and I found using sum() as a solution on the forums. Are you getting scalar returns?

Reply:
>I was having problems with th.sum

What was the issue exactly?
(I would prefer using `th.sum()` if possible as `sum()` should be calling it at some point internally anyway)

>I'm seeing the expected returns, no?

Looks good for that example, but best is to compare inside the `train()` method of an algorithm (e.g. ppo).